### ISSUE TITLE
Fullscreen video always freezing in Wayland session,  sound still playing (Fedora 40, Gnome 46, Nvidia GTX 1060)

### ISSUE THREAD

**Issue by encomosxii** [mpv-player/mpv#14385]
### mpv Information

```bash
mpv v0.38.0 Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
libplacebo version: v6.338.2
FFmpeg version: n7.0.1
FFmpeg library versions:
   libavutil       59.8.100
   libavcodec      61.3.100
   libavformat     61.1.100
   libswscale      8.1.100
   libavfilter     10.1.100
   libswresample   5.1.100
```

### Other Information

- Linux version: 
NAME="Fedora Linux"
VERSION_CODENAME=""
PRETTY_NAME="Fedora Linux 40 (Workstation Edition)"
CPE_NAME="cpe:/o:fedoraproject:fedora:40"
DEFAULT_HOSTNAME="fedora"

- Kernel Version:
Linux 6.8.11-300.fc40.x86_64

- GPU Model:
07:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP106 [GeForce GTX 1060 6GB] [10de:1c03] (rev a1)

- Mesa/GPU Driver Version:
OpenGL:
OpenGL version string: 4.6.0 NVIDIA 550.90.07

  Vulkan:
  DRM kernel driver 'nvidia-drm' in use. NVK requires nouveau.
  ERROR: [../src/nouveau/vulkan/nvk_physical_device.c:1109] Code 0 : VK_ERROR_INCOMPATIBLE_DRIVER
	driverInfo                                           = 550.90.07
	driverInfo                                           = Mesa 24.1.1 (LLVM 18.1.6)

- Window Manager and Version: 
Gnome 46

- Source mpv:
Flathub and Fedora RPM

- Introduced in version:

### Reproduction Steps

When switching to fullscreen (not maximise), movie freezes after a few moments, sound still playing for some time, up to 2-3 minutes sometimes, than mpv crashes.
- Screen freezes **only under Wayland** not X. 
- No error message via terminal. 
- Flatpak and RPM show same behavior. 
- Containerformat doesen't matter, mkv is my preferred container with h264 and h265 encoded material.
- Error occurs since months. Used Fedora 39 and Tumbleweed before, Gnome and KDE, everywhere the same problem with mpv.
- Tried no-config and different hwdec: vaapi, vdpau and nvdec. Screen keeps freezing.

Additional information that maybe help?
This also happens in Celluloid (0.26 Flatpak and RPM) which is based on mpv. Screen freeze occurs there too under same conditions, but the freeze can be prevented there by forcing the controls not to fade out (mouseover).

Every video player has problems with fullscreen under Wayland on my machine: 
Clapper is freezing too after a few moments with sound still playing, just like mpv. 
Plex starts stuttering in fullscreen after a few moments. 
VLC runs fine when X/X11 is used for video output, switching to OpenGL leads to stuttering video just like Plex. 

Because of the sample file for output.txt, Big Buck Bunny from Blender: [big_buck_bunny_480p_h264.mov](https://download.blender.org/peach/bigbuckbunny_movies/big_buck_bunny_480p_h264.mov)

### Expected Behavior

Video playback

### Actual Behavior

Video freezing

### Log File

[output.txt](https://github.com/user-attachments/files/15891604/output.txt)

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by Dudemanguy** [comment]
>Every video player has problems with fullscreen under Wayland on my machine

This is just a bug with nvidia drivers on wayland judging by your description. Nothing we can do here.

**Comment by spaceman7777** [comment]
Fix in link might be related. Poster removed the vulkan-nouveau packages, and the system appears to have reverted to correctly using the nvidia vulkan implementation. (I'm assuming you're using official nvidia drivers rather than nouveau.) You shouldn't need nouveau around to use nvidia drivers, and it is generally encouraged to blacklist the nouveau kernel module if they are still needed on the system for package management reasons.

Good luck :)

https://forum.endeavouros.com/t/error-with-lutris-drm-kernel-driver-nvidia-drm-in-use-nvk-requires-nouveau/53845/3

**Comment by encomosxii** [comment]
Thanks for your help Dudemanguy and spaceman7777! By the way, Noveau is deactivated, kernel modules are blacklisted.

Dudemanguy is right, seems to be a nvidia driver or maybe wayland and... problem. I was able to identify the problem.

I'm using a "multi monitor" setup. I use screen mirroring, not the extend screen mode for this setup. Screen mirroring causes all videoplayers to freeze and crash in fullscreen mode when using wayland.

The second monitor is my TV/Surround receiver which passes the TV signal through. 
I noticed since years different problems over many distributions with multi monitor setup. For example the second monitor, the TV, connected directly or via passthrough will be set to wrong refresh rate and resolution by the system. 
Nothing can prevent this, after a few reboots every config will be messed up again. 
Monitors.xml will be rebuild with the wrong settings, no matter how often you correct them. 
Setting a nvidia.conf file under /etc/x11 won't help, because all settings regarding monitor resolution and refresh rate will always be ignored.

The never ending story, wayland, xorg, nvidia, multi monitor... But fortunatly for all linux users, the real problems always have been in the focus. For example desktop icons in Gnome or the announcement of a new DE every week.....

**Comment by encomosxii** [comment]
Wayland Nvidia related problem. Setting a multi monitor setup to screen mirroring causes the problem.

**Comment by bruno-srio** [comment]
Just leaving this here in case it helps someone.

I stumbled upon this issue while trying to fix a similar problem on my system. I’m not involved with this project, I just landed here through search.

If you have fractional scaling enabled, try switching back to 100% scale and using Accessibility > Large Text instead. That solved the full-screen and video freezing issues for me (in all apps and websites).

Fedora 42 with GNOME 48 here.
